types/asana/asana-tests.ts

```diff
@@ -124,19 +155,251 @@ client.tasks.update('task_gid', {
+    external: {
+        data: "A blob of information",
+        gid: "external_identifier",
+    },
     html_notes: '<b>some html</b>',
     liked: true,
     name: 'some task name',
     notes: 'some description',
     parent: 'some parent task gid',
+    start_at: 'some_time',
     start_on: 'some_date',
     workspace: 'some_workspace_gid',
 }).then((task) => console.log(task.name));
+
+// redacted response from client.tasks.getTask('gid here') with no options passed,
+// from a Premium-level account.
+const _returnedTask: asana.resources.Tasks.Type = {
+    gid: "123",
+    assignee: {
+        gid: "456",
+        name: "person name",
+        resource_type: "user"
+    },
+    assignee_status: "inbox",
+    assignee_section: {
+        gid: "789",
+        name: "section name",
+        resource_type: "section"
+    },
+    completed: false,
+    completed_at: null,
+    created_at: "2022-02-19T14:39:02.472Z",
+    custom_fields: [
+        {
+            gid: "123",
+            enabled: true,
+            name: "upvotes",
+            number_value: -293,
+            precision: 0,
+            created_by: {
+                gid: "456",
+                name: "person name",
+                resource_type: "user"
+            },
+            display_value: "-293",
+            resource_subtype: "number",
+            resource_type: "custom_field",
+            type: "number"
+        }
+    ],
+    due_at: null,
+    due_on: null,
+    followers: [
+        {
+            gid: "123",
+            name: "person name",
+            resource_type: "user"
+        }
+    ],
+    hearted: false,
+    hearts: [],
+    liked: false,
+    likes: [],
+    memberships: [
+        {
+            project: {
+                gid: '456',
+                name: 'project name',
+                resource_type: 'project',
+            },
+            section: {
+                gid: '789',
+                name: 'section name',
+                resource_type: 'section',
+            },
+        },
+    ],
+    modified_at: '2022-05-29T23:33:19.232Z',
+    name: 'name here',
+    notes: 'notes here',
+    num_hearts: 0,
+    num_likes: 0,
+    parent: null,
+    permalink_url: 'https://app.asana.com/0/456/123',
+    projects: [
+        {
+            gid: '123',
+            name: 'project name',
+            resource_type: 'project',
+        },
+    ],
+    resource_type: 'task',
+    start_at: null,
+    start_on: null,
+    tags: [],
+    resource_subtype: 'default_task',
+    workspace: {
+        gid: '123',
+        name: 'workspace name',
+        resource_type: 'workspace',
+    },
+};
+
+// redacted response from client.tasks.getTask('gid here', {opt_fields: ['gid']})
+const _minimalReturnedTask: asana.resources.Tasks.Type = {
+    gid: '123',
+};
+
+// redacted response from client.tasks.getTask('gid here', {opt_fields: ['name']})
+const _secondMinimalReturnedTask: asana.resources.Tasks.Type = {
+    gid: '123',
+    name: 'name here',
+};
```

Comment:
The Asana API allows you to specify which fields get returned (with the apparent exception of gid, which seems to be required).  

As a result of this, I had to mark a lot of existing "required" fields on output as optional.  This is going to break some code on testing.  For instance, I had to change some code of my own this way:

```typescript
export const formatTask = (task: Asana.resources.Tasks.Type) => {
  if (task.memberships == null) {
    throw new Error('Memberships required to format!');
  }
  // now free to use task.membership
}
```

If there's a better way to handle this situation, I'm all ears.